[BUG]: run all was not configured in "multiple" section of config

### Discussed in https://github.com/codeceptjs/CodeceptJS/discussions/3803

<div type='discussions-op-text'>

<sup>Originally posted by **clundstedt-m** August 10, 2023</sup>
We are currently trying to run our tests in multiple browsers and eventually run those over our Lambdatest instance, nonetheless we haven't been able to even run them "locally" in multiple browsers nor the less at Lambdatest as when we try to run them using the **`run-workers |number| all`**, we are getting the error that I wrote in the discussion title, so maybe (_most likely_) we are missing some pre-requisites to be fulfilled prior to be able to run the actual _multiple_ feature?

The command that we are running is:
`npx codeceptjs run-workers 3 all -c codecept.conf.POCCrossBrowser.ts --features --grep @CrossBrowserPOC`

The specifics of the error are the following:
```
Could not load helper multiple from module './helper/multiple':
Cannot find module './helper/multiple'
Require stack:
- /Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/container.js
- /Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/codecept.js
- /Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/bin/codecept.js
Error: Cannot find module './helper/multiple'
Require stack:
- /Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/container.js
- /Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/codecept.js
- /Users/christian.lundstedt/tr-qa-automation/node_modules/codeceptjs/bin/codecept.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1070:15)
    at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/Users/christian.lundstedt/tr-front-qa-automation/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
    at Function.Module._load (node:internal/modules/cjs/loader:923:27)
    at Module.require (node:internal/modules/cjs/loader:1137:19)
    at require (node:internal/modules/helpers:121:18)
    at createHelpers (/Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/container.js:173:23)
    at Function.create (/Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/container.js:44:25)
    at Codecept.init (/Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/codecept.js:57:15)
    at initializeCodecept (/Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/workers.js:26:12)
    at new Workers (/Users/christian.lundstedt/tr-front-qa-automation/node_modules/codeceptjs/lib/workers.js:214:21)
```

Our current .conf file looks like this (_it includes the current capabilities for Lambdatest_ even if it doesn't/shouldn't really matter):

```typescript
import * as dotenv from 'dotenv';
dotenv.config();
import { merge } from 'lodash';
import {defaultConfig} from '@forwardthinking/codeceptjs-helpers'


const webConfig = {
  helpers: {
    Playwright: {
      url: 'https://demo.applitools.com/',
      show: true,
      waitForNavigation: 'networkidle',
      desiredCapabilities: {}
    },
    multiple: {
      profile1: {
        browsers: [
          {
            browser: "chrome",
            desiredCapabilties: {
              'browserName': 'Chrome',
              'browserVersion': '114.0',
              'LT:Options': {
                  'platform': 'Windows 10',
                  'build': 'LT Poc with Playwright',
                  'name': 'Chrome - LT Test',
                  'user': process.env.LT_USERNAME,
                  'accessKey': process.env.LT_KEY,
              }
            }
          },
          {
            browser: "firefox",
            desiredCapabilties: {
              'browserName': 'pw-firefox',
              'browserVersion': 'latest',
              'LT:Options': {
                  'platform': 'Windows 10',
                  'build': 'LT Poc with Playwright',
                  'name': 'Firefox - LT Test',
                  'user': process.env.LT_USERNAME,
                  'accessKey': process.env.LT_KEY,
            }
          }
          },
        ]
      },
      profile2: {
        browsers: [
          {
            browser: "safari",
            desiredCapabilties: {
              'browserName': 'pw-chromium',
              'browserVersion': 'latest',
              'LT:Options': {
                  'platform': 'MacOS Big sur',
                  'build': 'LT Poc with Playwright',
                  'name': 'Chromium MacOS - LT Test',
                  'user': process.env.LT_USERNAME,
                  'accessKey': process.env.LT_KEY,
            }
            }
          }
        ]
      }, 
    //Helper API: Used to execute pre-conditions for specific API's
    REST: {
      endpoint: 'https://swapi.dev/api',
      timeout: 10000
    },
    //recommended Helper to work with API with specific JSON responses
    JSONResponse: {},
  },

  plugins: {
  //disabled for POC purposes
    cucumberJsonReporter: { enabled: false},
    stats: { enabled: false },
  },
}
}

const config = merge(defaultConfig,webConfig);
export {config}
``` 

Any guidance for this issue would be greatly appreciated as we cannot find any information or similar issues around the only and even have other people that got it to work without any hassle, like this one -- https://codecept.discourse.group/t/running-tests-across-multiple-browsers-in-playwright/761/2

Thanks in advance! </div>

--- a/lib/command/run-multiple/collection.js
+++ b/lib/command/run-multiple/collection.js
@@ -47,14 +47,26 @@ class Collection {
    */
   prepareRuns(selectedRuns, config) {
     selectedRuns.forEach((selectedRun) => {
-      const [runName] = selectedRun.split(':');
-      const runConfig = config.multiple[runName];
+      const runConfig = [];
+      let runName = [];
 
-      if (!runConfig) {
-        throw new Error(`run ${runName} was not configured in "multiple" section of config`);
+      if (selectedRun === 'all') {
+        Object.keys(config.multiple).forEach(name => {
+          runName.push(name);
+          runConfig.push(config.multiple[name]);
+        });
+      } else {
+        runName = selectedRun.split(':')[0];
+        runConfig.push(config.multiple[runName]);
       }
 
-      this.addRun(createRun(runName, runConfig));
+      for (let i = 0; i < runConfig.length; i++) {
+        if (!runConfig[i]) {
+          throw new Error(`run ${runName[i]} was not configured in "multiple" section of config`);
+        }
+
+        this.addRun(createRun(runName[i], runConfig[i]));
+      }
     });
 
     return this;

--- a/lib/command/run-multiple/collection.js
+++ b/lib/command/run-multiple/collection.js
@@ -48,7 +48,7 @@ class Collection {
   prepareRuns(selectedRuns, config) {
     selectedRuns.forEach((selectedRun) => {
       const runConfig = [];
-      let runName = [];
+      const runName = [];
 
       if (selectedRun === 'all') {
         Object.keys(config.multiple).forEach(name => {
@@ -56,8 +56,8 @@ class Collection {
           runConfig.push(config.multiple[name]);
         });
       } else {
-        runName = selectedRun.split(':')[0];
-        runConfig.push(config.multiple[runName]);
+        runName.push(selectedRun.split(':')[0]);
+        runConfig.push(config.multiple[runName[0]]);
       }
 
       for (let i = 0; i < runConfig.length; i++) {

